Source/Evolve/WarpXEvolve.cpp

```diff
@@ -305,6 +305,8 @@ WarpX::OneStep_nosub (Real cur_time)
     doQEDEvents();
 #endif
 
+    mypc->doResampling(istep[0]+1);
```

Comment:
Why do we need to pass `istep[0]+1` instead of `istep[0]` ? I would suggest to add a short comment to explain why this is necessary...

Reply:
There's a difference of one between the value of `step` (or `istep[0]`) in the code and the value that's actually printed to the user. see https://github.com/ECP-WarpX/WarpX/blob/ec64b868cde2e555adb3475a9a521ce08cd73c21/Source/Evolve/WarpXEvolve.cpp#L55

So the +1 is needed so that if the user asks for a resampling at step 5, he actually sees it happening at step 5. (this trick is actually used several times in Evolve, see for instance https://github.com/ECP-WarpX/WarpX/blob/ec64b868cde2e555adb3475a9a521ce08cd73c21/Source/Evolve/WarpXEvolve.cpp#L65 or https://github.com/ECP-WarpX/WarpX/blob/ec64b868cde2e555adb3475a9a521ce08cd73c21/Source/Evolve/WarpXEvolve.cpp#L208 )

I agree that it's not ideal. I will add some comments (or maybe we can add a variable named `user_step` or similar that is used everytime we currently use `step +1` ?).

Reply:
Ah ok! Now it's clear... I think that a short, one-line, comment would be enough (adding a variable `user_step` in `WarpX` would probably require some discussion with the other developers).